diff -uNrp opensips-1.11.8.orig/modules/auth_db/authdb_mod.c opensips-1.11.8/modules/auth_db/authdb_mod.c
--- opensips-1.11.8.orig/modules/auth_db/authdb_mod.c	2016-06-17 21:33:16.000000000 +0700
+++ opensips-1.11.8/modules/auth_db/authdb_mod.c	2016-09-29 22:22:03.106184117 +0700
@@ -98,6 +98,7 @@ str pass_column             = {PASS_COL,
 str pass_column_2           = {PASS_COL_2, PASS_COL_2_LEN};
 
 
+int join                    = 0;
 int calc_ha1                = 0;
 int use_domain              = 0; /* Use also domain when looking up in table */
 
@@ -131,6 +132,7 @@ static param_export_t params[] = {
 	{"password_column_2", STR_PARAM, &pass_column_2.s    },
 	{"calculate_ha1",     INT_PARAM, &calc_ha1           },
 	{"use_domain",        INT_PARAM, &use_domain         },
+	{"join",              INT_PARAM, &join               },
 	{"load_credentials",  STR_PARAM, &credentials_list   },
 	{"skip_version_check",INT_PARAM, &skip_version_check },
 	{0, 0, 0}
diff -uNrp opensips-1.11.8.orig/modules/auth_db/authdb_mod.h opensips-1.11.8/modules/auth_db/authdb_mod.h
--- opensips-1.11.8.orig/modules/auth_db/authdb_mod.h	2016-06-17 21:33:16.000000000 +0700
+++ opensips-1.11.8/modules/auth_db/authdb_mod.h	2016-09-29 22:22:02.859515867 +0700
@@ -43,7 +43,8 @@ extern str pass_column;     /* 'password
 extern str pass_column_2;   /* Column containing HA1 string constructed
 			     * of user@domain username
 			     */
-
+extern int join;              /* if set to 1, username and domain will be joint to perform queries
+                                 Thus this requires use_domain set to 0 */
 extern int calc_ha1;          /* if set to 1, ha1 is calculated by the server */
 extern int use_domain;        /* If set to 1 then the domain will be used when selecting a row */
 
diff -uNrp opensips-1.11.8.orig/modules/auth_db/authorize.c opensips-1.11.8/modules/auth_db/authorize.c
--- opensips-1.11.8.orig/modules/auth_db/authorize.c	2016-06-17 21:33:16.000000000 +0700
+++ opensips-1.11.8/modules/auth_db/authorize.c	2016-09-29 22:22:01.369506297 +0700
@@ -94,10 +94,27 @@ static inline int get_ha1(struct usernam
 	VAL_STR(vals).s = _username->user.s;
 	VAL_STR(vals).len = _username->user.len;
 
+	str u_n_d;
 	if (_username->domain.len) {
 		VAL_STR(vals + 1) = _username->domain;
+		if (join == 1) {
+			u_n_d.len = _username->user.len + _username->domain.len + 1;
+			u_n_d.s = pkg_malloc(u_n_d.len);
+			sprintf(u_n_d.s, "%.*s@%.*s",_username->user.len, _username->user.s, _username->domain.len, _username->domain.s);
+		}
 	} else {
 		VAL_STR(vals + 1) = *_domain;
+                if (join == 1) {
+			u_n_d.len = _username->user.len + _domain->len + 1;
+			u_n_d.s = pkg_malloc(u_n_d.len);
+			sprintf(u_n_d.s, "%.*s@%.*s",_username->user.len, _username->user.s, _domain->len, _domain->s);
+		}
+	}
+
+	if (join == 1) {
+		VAL_STR(vals).s = u_n_d.s;
+		VAL_STR(vals).len = u_n_d.len;
+		use_domain = 0;
 	}
 
 	if (auth_dbf.use_table(auth_db_handle, _table) < 0) {
